src/overlay.cpp

```diff
@@ -639,57 +638,35 @@ void init_gpu_stats(uint32_t& vendorID, uint32_t reported_deviceID, overlay_para
             metrics_path = gpu_metrics_path;
             SPDLOG_DEBUG("Using gpu_metrics");
          }
-#ifdef HAVE_LIBDRM_AMDGPU
-         else {
-            int idx = -1;
-            //TODO make neater
-            int res = sscanf(path.c_str(), "/sys/class/drm/card%d", &idx);
-            std::string dri_path = "/dev/dri/card" + std::to_string(idx);
-
-            if (!params.enabled[OVERLAY_PARAM_ENABLED_force_amdgpu_hwmon] && res == 1 && amdgpu_open(dri_path.c_str())) {
-               vendorID = 0x1002;
-               using_libdrm = true;
-               getAmdGpuInfo_actual = getAmdGpuInfo_libdrm;
-               amdgpu_set_sampling_period(params.fps_sampling_period);
-
-               SPDLOG_DEBUG("Using libdrm");
-
-               // fall through and open sysfs handles for fallback or check DRM version beforehand
-            } else if (!params.enabled[OVERLAY_PARAM_ENABLED_force_amdgpu_hwmon]) {
-               SPDLOG_ERROR("Failed to open device '/dev/dri/card{}' with libdrm, falling back to using hwmon sysfs.", idx);
-            } else if (params.enabled[OVERLAY_PARAM_ENABLED_force_amdgpu_hwmon]) {
-               SPDLOG_DEBUG("Using amdgpu hwmon");
-            }
-         }
-#endif
-
          path += "/device";
-         if (!amdgpu.busy)
-            amdgpu.busy = fopen((path + "/gpu_busy_percent").c_str(), "r");
          if (!amdgpu.vram_total)
             amdgpu.vram_total = fopen((path + "/mem_info_vram_total").c_str(), "r");
          if (!amdgpu.vram_used)
             amdgpu.vram_used = fopen((path + "/mem_info_vram_used").c_str(), "r");
 
+         if (!metrics_path.empty())
+            break;
+
+         if (!amdgpu.busy)
+            amdgpu.busy = fopen((path + "/gpu_busy_percent").c_str(), "r");
          path += "/hwmon/";
-         string tempFolder;
-         if (find_folder(path, "hwmon", tempFolder)) {
+         auto dirs = ls(path.c_str(), "hwmon", LS_DIRS);
+         for (auto& dir : dirs) {
             if (!amdgpu.core_clock)
-               amdgpu.core_clock = fopen((path + tempFolder + "/freq1_input").c_str(), "r");
+               amdgpu.core_clock = fopen((path + dir + "/freq1_input").c_str(), "r");
             if (!amdgpu.memory_clock)
-               amdgpu.memory_clock = fopen((path + tempFolder + "/freq2_input").c_str(), "r");
+               amdgpu.memory_clock = fopen((path + dir + "/freq2_input").c_str(), "r");
             if (!amdgpu.temp)
-               amdgpu.temp = fopen((path + tempFolder + "/temp1_input").c_str(), "r");
+               amdgpu.temp = fopen((path + dir + "/temp1_input").c_str(), "r");
             if (!amdgpu.power_usage)
-               amdgpu.power_usage = fopen((path + tempFolder + "/power1_average").c_str(), "r");
+               amdgpu.power_usage = fopen((path + dir + "/power1_average").c_str(), "r");
 
-            vendorID = 0x1002;
             break;
```

Comment:
Tangling `break`?

Reply:
Yup should be after the `}` - fix coming in a moment.